library/core/src/char/methods.rs

```diff
@@ -457,8 +463,8 @@ impl char {
     /// ```
     #[stable(feature = "char_escape_debug", since = "1.20.0")]
     #[inline]
-    pub fn escape_debug(self) -> EscapeDebug {
-        self.escape_debug_ext(true)
+    pub fn escape_debug(self, is_char: bool) -> EscapeDebug {
```

Comment:
This function has been part of the stable API since Rust 1.20. We can't change it.

I suppose this function should keep escaping both `\"` and `\'`, to make sure we don't break anything.

Reply:
Hmm, this is unfortunate, because this function doesn't make a lot of sense, as `str` and `char` have different escape rules.

I wonder if we could repurpose this for only `str` or only `char`, and have another function for the other one?

I'm curious what the use case for this function is (other than the internal uses). #35068 doesn't give a lot of context. Any ideas on why/how this function is used by the downstream?

Reply:
Can't you just call `self.escape_debug_ext(true, true)` in `char::escape_debug` and `self.escape_debug_ext(true, false)` in `str::escape_debug`? Am I missing something obvious?

Reply:
I guess another option would be using another function internally and in the libs, and leaving this function as-is.

The documentation says 

> This will escape the characters similar to the `Debug` implementations of `str` or `char`

If we did that, `str` and `char` would no longer be using this function. Perhaps that's OK as this says "similar" and not "exactly".

I wonder if this is an option, but I think we could also deprecate this, as this function is basically never correct. If you use this for printing chars then you'll escape `"` incorrectly. If you use it to print string then you'll escape `'`s incorrectly.

Reply:
> Can't you just call self.escape_debug_ext(true, true) in char::escape_debug and self.escape_debug_ext(true, false) in str::escape_debug? Am I missing something obvious?

Ah, I completely missed that. I'll try that now.

Reply:
Thanks for the suggestion @fmease , it worked nicely. Resolving.

Reply:
I don't think we should change the behaviour of `char::escape_debug()` or `str::escape_debug()`. We can't know whether those will be printed between single or double quotes. We can only modify this behaviour in the places where we know for sure which surrounding quotes are used because we print them ourself (e.g. the Debug implementations).

Reply:
OK, reverted `escape_debug` changes.